tool/tctl/common/collection_test.go

```diff
@@ -431,3 +437,44 @@ func makeTestLabels(extraStaticLabels map[string]string) map[string]string {
 	maps.Copy(labels, extraStaticLabels)
 	return labels
 }
+
+// This test makes sure we marshal and unmarshal proto-based Resource153 properly.
+// We had a bug where types.Resource153 implemented by protobuf structs were not
+// marshaled properly (they should be marshaled using protojson). This test
+// checks we can do a round-trip with one of those proto-struct resource.
+func TestRoundTripProtoResource153(t *testing.T) {
```

Comment:
The test name should mention Resource153ToLegacyV2, which it does rely on (unless I'm reading it wrong?).

It should also, ideally, compare the behavior with the regular Resource153ToLegacy which purposefully does not use protojon.

Reply:
Done, added a failing test case.